agent/testagent.go

```diff
@@ -369,6 +374,43 @@ func (a *TestAgent) consulConfig() *consul.Config {
 	return c
 }
 
+// fakeRPCDelegate wraps the delegate interface, overriding the RPC method,
+// to support replacing RPC calls with fakes.
+type fakeRPCDelegate struct {
+	delegate
+	endpoints     map[string]string
+	endpointsLock sync.RWMutex
+}
+
+func (d *fakeRPCDelegate) RPC(method string, args interface{}, reply interface{}) error {
+	d.endpointsLock.RLock()
```

Comment:
I suspect this `endpointsLock` may be safe to remove, now that the code is only in the `TestAgent`, but I was not confident, so I left it for now.